lib/collection/src/shards/local_shard.rs

```diff
@@ -206,14 +207,27 @@ impl LocalShard {
             );
         }
 
-        for handler in load_handlers {
-            let segment_opt = handler.join().map_err(|err| {
-                CollectionError::service_error(format!(
-                    "Can't join segment load thread: {:?}",
-                    err.type_id()
-                ))
-            })??;
-            if let Some(segment) = segment_opt {
+        {
+            // TODO: Is it fine to hold read-lock on `collection_config` here? 🤔
+            let collection_config = collection_config.read().await;
```

Comment:
We already have a read lock at the top of the function. I'll push a commit on top of this branch for it.